packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+  }
+
+  public async shutdown(): Promise<void> {
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+    }
+  }
+
+  async registerWebServices(
+    app: Express,
+    //wsApi: SocketIoServer,
+  ): Promise<IWebServiceEndpoint[]> {
+    //const { web3 } = this;
+    //const { logLevel } = this.options;
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+
+    // wsApi.on("connection", (socket: SocketIoSocket) => {
+    //   this.log.debug(`New Socket connected. ID=${socket.id}`);
+
+    //   socket.on(WatchBlocksV1.Subscribe, () => {
+    //     new WatchBlocksV1Endpoint({ web3, socket, logLevel }).subscribe();
+    //   });
+    // });
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+    const endpoints: IWebServiceEndpoint[] = [];
+    {
+      const endpoint = new RunTransactionEndpoint({
+        connector: this,
+        logLevel: this.options.logLevel,
+      });
+      endpoints.push(endpoint);
+    }
+    // {
+    //   const endpoint = new InvokeContractEndpoint({
+    //     connector: this,
+    //     logLevel: this.options.logLevel,
+    //   });
+    //   endpoints.push(endpoint);
+    // }
+
+    {
+      const opts: IGetPrometheusExporterMetricsEndpointV1Options = {
+        connector: this,
+        logLevel: this.options.logLevel,
+      };
+      const endpoint = new GetPrometheusExporterMetricsEndpointV1(opts);
+      endpoints.push(endpoint);
+    }
+    this.endpoints = endpoints;
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ledger-connector-iroha`;
+  }
+
+  public async getConsensusAlgorithmFamily(): Promise<
+    ConsensusAlgorithmFamily
+  > {
+    return ConsensusAlgorithmFamily.Authority;
+  }
+  public async hasTransactionFinality(): Promise<boolean> {
+    const currentConsensusAlgorithmFamily = await this.getConsensusAlgorithmFamily();
+
+    return consensusHasTransactionFinality(currentConsensusAlgorithmFamily);
+  }
+  //...  spread
+  //pass in the params
+  public async transact(
+    req: RunTransactionRequest, //string + array<any>
+  ): Promise<RunTransactionResponse> {
+    const { baseConfig } = req;
+    if (
+      !baseConfig ||
+      !baseConfig.privKey ||
+      !baseConfig.creatorAccountId ||
+      !baseConfig.irohaHost ||
+      !baseConfig.irohaPort ||
+      !baseConfig.quorum ||
+      !baseConfig.timeoutLimit
+    ) {
+      console.error("No data here!");
+      return { transactionReceipt: "Some fields in baseConfig is undefined" };
+    }
+    const irohaHostPort = `${baseConfig.irohaHost}:${baseConfig.irohaPort}`;
+    const adminPriv =
```

Comment:
@hxlaf We cannot have hardcoded keys in the connector, everything must come from either the constructor args or the request object otherwise it's not reusable.

Reply:
Hardcoded key removed! Thank you!